### ISSUE TITLE
Allow Admins to Identify & Purge Old Projects via the UI

### ISSUE THREAD

**Issue by b-f-chan** [overture-stack/enrolment#109]
**Requirements:**

* Provide a UI for admins to identify & purge old projects from the system/database
* Can be an admin-only screen or control in existing Enrolment UI

**Out-of-Scope:**

* For now assume admin will determine manually via policy what is considered "old"
* *Future enhancement* - Allow admins to mark or flag specific projects for purging - Can allow a flag or label, or can consider time-based configuration
* *Future enhancement* - Auto-purging based on configurable time period

**Context:**

* Due to bugs in enrolment and perhaps suboptimal design of the user experience, many test projects are often created in production to make sure things work.  Need an easy way to clean them up.

**Comment by jmimico** [comment]
Any update? This is a really important feature for us.

**Comment by b-f-chan** [comment]
No update on this one either.  Let's take the priority ones you feel are impacting your time and/or user efficiency most and make a case with @Fgerthoffert @christinayung.

**Comment by mbrazas** [comment]
Can I also add a check box on the registration page that the “PI acknowledges that their usage of the Collaboratory will be invoiced monthly, and that payment is required within 30 days.” ?
I want to bring payment upfront if we are not going down the route of seeking credit card information upfront.
Michelle

--
Michelle D. Brazas, Ph.D.
Senior Program Manager, Adaptive Oncology
Project Manager, Cancer Genome Collaboratory

To learn about our Informatics research resources, click here<http://oicr.on.ca/collaborative-research-resources#hide2>
Ontario Institute for Cancer Research
MaRS Centre
661 University Avenue, Suite 510
Toronto, Ontario, Canada M5G 0A3

Tel: 416-673-8502
Cell: 416-357-6687
FAX: 416-977-7446
Toll-free: 1-866-678-6427 ext. 8502
Email: michelle.brazas@oicr.on.ca<x-msg://1768/michelle.brazas@oicr.on.ca> <http://michelle.brazas@oicr.on.ca<http://michelle.brazas@oicr.on.ca/>>
www.oicr.on.ca<http://www.oicr.on.ca/>

**Comment by b-f-chan** [comment]
Hi @mbrazas, good suggestion, I've captured this as a separate requirement in #116.

**Comment by jmimico** [comment]
Manual steps for deleting projects documented @ https://wiki.oicr.on.ca/display/Collabor/Enrolment+Application+Database+tips+and+tricks

**Comment by caravinci** [comment]
Thanks for that document, @jmimico! Missing from those steps is an additional correlation between applications and billing contacts.
I've automated the purging of those as well, and will be adding the relationship in your document.

We will need to clear any orphaned contacts in prod sometime later; but at least once this automation is applied in that environment, no new orphans will be created.

**Comment by caravinci** [comment]
The approach I've taken was to allow purging a project once it's been either `terminated` or `denied`, and the option will only be offered to staff members. This ensures following the user feedback process (e.g. email notifications, etc.) even for stale projects, before removing the entries from the database.

There's no email notification for that last step; and the project, along with all related entities, will simply be unlisted in the UI.

**Comment by caravinci** [comment]
Running some tests, we've noticed some remaining orphaned projects.
Upon deeper research, it seems they were remainders of some failed Cascading calls removing billing contacts, so I've implemented additional staff measures.
- The Applications table will now show staff members all the projects in the database, including orphaned ones.
- The server will be able to provide additional logging to help debugging these events.
- The purging logic is much more robust, and handles the possible edge cases consistently.

All these changes are available to test at lab @b-f-chan @jmimico
